Adicionando ambiente de desenvolvimento

jackmakiyama opened the pull request:
Fixes #6;

Ambiente de desenvolvimento padronizado com o Docker e orquestrado com o
docker-compose, contendo: PHP7, Postgres 9.6 e nginx.

Commit:
Adicionando ambiente de desenvolvimento

Ambiente de desenvolvimento padronizado com o Docker e orquestrado com o
docker-compose, contendo: PHP7, Postgres 9.6 e nginx.

Commit:
Formatando Dockerfile

Commit:
Adicionando Makefile para instalação do composer

Adicionando Makefile que instala o composer via docker usando o
container `composer/composer`.

jackmakiyama commented:
Para instalar as dependências do composer:

```
$ make composer.install
```

Commit:
Mudando tabs para espaços no Dockerfile.

jackmakiyama commented:
@EvandroMohr @f0ntana @enieber review please.

enieber commented on docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+version: '2'
```
o que seria esse version 2? tentei rodar no meu docker-compose mas deu erro

jackmakiyama replied:
é a versão do cocker.composer.yml, [veja mais aqui](https://medium.com/@giorgioto/docker-compose-yml-from-v1-to-v2-3c0f8bb7a48e#.5t8zm9tvq).

jackmakiyama replied:
Atualize seu Docker se puder.

enieber reviewed (commented):
ficou legal essa parte de rodar o composer com o make... mas tem que instalar alguma coisa para rodar esse makefiel ?

jackmakiyama commented:
> ficou legal essa parte de rodar o composer com o make... mas tem que instalar alguma coisa para rodar esse makefiel ?

Rapaz, creio que não precisa de nada, a maioria dos unix-like tem, na duvida: `make --version` no terminal.

enieber commented:
cara ficou legal, mas não consegui testar por causa do tamanho, minha internet não ajuda..

jackmakiyama commented:
> cara ficou legal, mas não consegui testar por causa do tamanho, minha internet não ajuda..

😢

enieber commented:
@jackmakiyama sera que da para pegar imagens menores para facilitar a portabilidade do projeto?

EvandroMohr commented on docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+        volumes:
+          - postgres-data:/var/lib/postgresql/data
+        ports:
+          - "5432:5432"
```
Sugiro mudar a porta que ele sobe. pra não dar conflito caso já tenha outra instancia do postgresql rodando

EvandroMohr commented on docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+    web:
+        image: nginx:latest
+        ports:
+            - "80:80"
```
mudar porta padrão, para evitar conflito com outras instâncias

EvandroMohr reviewed (changes requested):
Alterar portas padrão para evitar conflito de ambiente

Commit:
Mudando portas do nginx e postgres

Mudando as portas para não dar conflito com alguma configuração
pre-existente do sistema operacional.
